gluten-ut/spark35/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala

```diff
@@ -92,6 +92,16 @@ class VeloxTestSettings extends BackendTestSettings {
     .exclude(
       "Process Infinity, -Infinity, NaN in case insensitive manner" // +inf not supported in folly.
     )
+  enableSuite[GlutenTryCastSuite]
+    .exclude(
+      "Process Infinity, -Infinity, NaN in case insensitive manner" // +inf not supported in folly.
+    )
+    .exclude("ANSI mode: Throw exception on casting out-of-range value to byte type")
```

Comment:
Would you add some comments on why these tests are excluded? I remember ANSI ON causes execution fallback, can the tests produce Spark expected result?

Reply:
These tests come from https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastWithAnsiOnSuite.scala and when the suite is run separately with `SQLConf.get.setConf(SQLConf.ANSI_ENABLED, true)` falls back but when run as part of `GlutenTryCastSuite` where the conf is not set does not give expected results. (Also, the tests fail without the code changes in this PR.)

Reply:
> the conf is not set does not give expected results

Thanks for providing the details. These tests are for ANSI OFF cases, right?

Reply:
These tests are for ANSI ON.
@acvictor, for these tests in TryCastSuite, exceptions are not thrown, but null results are returned? It should be an expected behavior for Velox cast.
I cannot figure out how these exception checks can pass in vanilla Spark, assuming exception is handled internally in TRY mode.

Reply:
@PHILO-HE I don't fully understand what you mean. Are you saying Velox behaviour is incorrect?

Reply:
@acvictor, please rebase the code. The core change looks good to me.

Reply:
Updated, thank you!